### PR TITLE
Configuration of config center url

### PR DESCRIPTION
**Pull request by wujimin**
1.read config center url from local config data, that include microservice.yaml/additional config/environment/command parameters
　2.discuss with config-cc, modify SPI interface, not use WatchedConfigurationSource directly, but extend it, add method: void init(Configuration localConfig), to avoid create another configration instance and reduce api of create configration coupling

### CODE REVIEW

File: foundations/foundation-config/src/main/java/io/servicecomb/config/ConfigUtil.java [hunk 1]
```diff
@@ -49,7 +45,7 @@
 public final class ConfigUtil {
     private static final Logger LOGGER = LoggerFactory.getLogger(ConfigUtil.class);
 
-    private static final String configCenterUrlKey = "cse.config.client.serverUri";
+    protected static final String configCenterUrlKey = "cse.config.client.serverUri";
```
Comment (liubao68) [inline foundations/foundation-config/src/main/java/io/servicecomb/config/ConfigUtil.java]: 对于final class，不应该有protected方法

Comment (wujimin) [inline foundations/foundation-config/src/main/java/io/servicecomb/config/ConfigUtil.java]: share for unit test.

**Comment by WillemJiang** [comment]
Hi jiming,  there are some changes on the ConfigUtil. Could you update the code and resolve the conflicts?

**Comment by wujimin** [comment]
resolved conflicts
and fixed a JAV-129 bug

**Pull request merged**